common/network-common/src/main/java/org/apache/spark/network/sasl/SaslClientBootstrap.java

```diff
@@ -88,9 +90,20 @@ public void doBootstrap(TransportClient client, Channel channel) {
           throw new RuntimeException(
             new SaslException("Encryption requests by negotiated non-encrypted connection."));
         }
-        SaslEncryption.addToChannel(channel, saslClient, conf.maxSaslEncryptedBlockSize());
+
+        if (conf.aesEncryptionEnabled()) {
+          // Generate a request config message to send to server.
+          AesConfigMessage configMessage = AesCipher.createConfigMessage(conf);
+          ByteBuffer buf = configMessage.encodeMessage();
+          client.sendRpcSync(buf, conf.saslRTTimeoutMs());
```

Comment:
Wait. Isn't this now sending the keys in plain text over the wire?

I commented before that using 3des to encrypt the keys is sub-optimal, but I didn't mean for you to remove the encryption altogether.